
\begin{abstract}
  This report present the work carried out during my Master's Thesis in the Megason Lab, Harvard Medical School, Boston MA, USA.
  This internship is the last step of my formation in the National Institute of Applied Sciences of Lyon (INSA), and at the University Claude Bernard Lyon 1 (UCLB), to get the Master of Sciences in Electrical engineering and Proceeds, option Systems and Images, and the Master of Engineering in Electrical Engineering and Computer Sciences.

  The work presented here is based on microscopy images processing. I have been working on three dimension plus time datasets representing a developing zebra fish embryo. 
  The laboratory employs a team of computer scientists in charge of development of a program for biological data visualization and processing :{\gofigure}\cite{refGofigure2}.
  The project carried out while in the Megason Lab intend to be integrated to this program.
  
\tableofcontents  
  
 
\end{abstract}


\chapter{Context}

\begin{figure}[htb]
\begin{center}
\leavevmode
\includegraphics[width=0.99\textwidth]{pictures/zebrafishPic}
\end{center}
\caption{Mature zebra fish: approximatively 2.5cm long (source: \href{http://en.wikipedia.org/wiki/File:Zebrafisch.jpg}{Wikipedia})}
\label{fig:ZebraPic}
\end{figure}

\section{Systems Biology}

With the completion of the genome projects~\cite{}, the stage is now set for a
more complete understanding of how animals are created. Biologists now have
access to the complete sequence of DNA that encodes the blueprint of life for
many animals including human, mouse, zebrafish, and fruit
fly~\cite{[2][3][4][5]} \TODO{refe?}The challenge is now understanding how this
code functions. The problem however is that the only thing that biologists can
currently reliably predict from the genomic code is which parts can be
transcribed and translated into protein. Biologists cannot accurately predict
when and where a protein will be expressed, how a protein will
function once it is created, and the interaction between expressed proteins that
allow them to form functional genetic circuits. Thus, although the genome
projects have given us the complete code for constructing many organisms, the
only part of the code biologists can currently read is the parts list. The
challenge is understanding how all these parts fit together to form molecular
circuits, how these circuits process information to regulate the behavior of
cells, and how the behavior of cells is orchestrated to generate functional form
as in \emph{embryogenesis}. This post-genomic endeavor has come to be called
\emph{systems biology}.\\

Although systems biology grew out of the “-omics” field which made heavy use of 
in vitro biochemical approaches (e.g. sequencing, microarrays, and proteomics),
\textit{in-vivo} imaging is becoming an increasingly powerful tool for
systems biology. \textit{In-vivo} imaging is advantageous over
biochemical approaches for doing systems biology:

\begin{itemize}
 \item Biological circuits function at the single cell level. Microscopic
imaging easily achieves this resolution while in-vitro techniques do not.
 \item Biological circuits function over time. With imaging, different
components of a biological circuit can be labeled using different colors of
fluorescent proteins and the dynamics of the circuit monitored non-invasively
with time-lapse fluorescent imaging as the circuit functions in an intact
system. In vitro biochemical approaches typically require the tissue to
be destroyed in order to be assayed which precludes longitudinal analysis.
 \item the quantitative amounts of components in a biological circuit are
important for its function. Fluorescent imaging can accurately quantitate the
levels of molecular components even at the protein level through the use of
fluorescent protein (e.g. GFP) fusions. Omic approaches are typically less
quantitative and focus at the DNA or RNA level which is less relevant.
 \item the anatomical context of biological circuits is essential for
determining their function; the use of spatial cues to generate different cell
types in different places is a fundamental aspect of development. In vivo
imaging can capture data from intact animals preserving its anatomical context.
In vitro approaches typically grind up the tissue to assay it, thus destroying
its anatomy.
\end{itemize}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Zebrafish: one sytem of high interest for Systems-Biology}

Zebrafish (Danio rerio) is a small, freshwater, tropical fish commonly available
in pet stores (see figure~\ref{}). It has become a popular model system for the
study of genetics and development in the last 2 decades and there are now
many labs worldwide that study it. There have been several
large-scale mutagenesis screens caried out in zebrafish and its genome has been
largely sequenced. Zebrafish has the advantages of fruit fly in that it
is amenable to forward genetic screens, but unlike fruit fly, zebrafish is a
vertebrate so is much more relevant to humans. Another huge advantage
of zebrafish is its suitability for imaging. Zebrafish embryos are transparent,
small, develop freely outside their mother, and develop directly from egg to
adult without any larval stages. This means that a zebrafish egg can be placed
under a microscope and continuously imaged throughout embryogenesis. This would
be impossible with a mouse which develop inside its mother, with a frog
(Xenopus) egg which is opaque, or with a fruit fly (Drosophila) which has
several larval/pupal stages and is opaque.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{In-toto imaging: one technique of interest for Systems-Biology}

The goal of in toto imaging is to image and track every single cell in a
developing tissue, organ, or eventually
whole embryo~\cite{megason2003digitizing}. This technique will provide
significant outcomes in a complete understanding of the cellular basis of
development through the construction of complete lineage trees\footnote{First
results on early zebrafish embryo~\cite{SCIENCE-PAPER} show the immense impact
on the scientific community of such a technique.}. There are several steps to in
toto imaging.\\

The embryos must first be labeled with “segmentation markers” which allow all
the cells to be segmented. Additionally embryos can be labeled with additional
markers to reveal RNA or protein expression. The next step is to acquire 3D+t
image sets using confocal or 2-photon fluorescent microscopy. And the final step
is processing the often very large image sets to segment out all the cell
trajectories and to extract quantitative, cell-centric data.

\subsection{Embryo Labeling}

The living embryos must be labeled with fluorescent markers and imaged with
time-lapse technique. Biologists must use vital labels such as Green
Fluorescent Protein (GFP), or other available Fluorescence Proteins (FPs) across
the visible spectrum. Since FPs are proteins rather than organically synthesized
small molecule dyes, FPs can be endogenously expressed by transgenic organisms
and are open to all the power of genetic engineering.\\

There are 2 basic uses of markers for in toto imaging: segmentation and
expression.
\begin{itemize}
 \item Segmentation markers allow the cells to be segmented in space,
across time, and across cell-division. Biologists, at the Megason Lab, typically
use a histone-FP of one color (e.g. histone-cerulean) and a membrane-localized
FP of another color (e.g. membrane-mCherry). This combination would generate
embryos in which every nuclei is green and all the cell membranes are red~(see
figure~\ref{}).
 \item Expression markers refer to whatever else is being marked such that it
can be digitized with in toto imaging. For example, transgenic fish can be made
that express a fluorescent protein wherever some gene is normally expressed
during development.
\end{itemize}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{Image acquisition}

\subsubsection{Requirements}

In toto imaging tracks cells during development only by the correspondence of
cells from frame to frame. This is quite different than traditional fate mapping
in developmental biology where only a small subset of cells is labeled at one
stage of development and the location of their progeny is observed at a
later stage. In toto imaging thus requires images that have high enough
resolution in space to resolve all the cells (this requires $\approx
1 \mu$m resolution) and high enough resolution in time to not lose track of
moving and dividing cells (this requires $\approx 2$ min time resolution). In
addition to high resolution, in toto imaging also requires complete coverage.
All the cells of interest must be continuously imaged in order to be tracked; if
cells leave the field of view then their lineage cannot be completely tracked.

\subsubsection{Equipment}

Achieving both high spatial and temporal resolution as well as complete coverage
can be achieved through the use of time-lapse confocal or 2-photon microscopy.
Confocal microscopy~(see figure~\ref{fig:ConfocalPrinciple}) uses a pinhole to
eliminate out of focus light whereas 2-photon microscopy only excites
fluorescence in the focal plane.

Thus both techniques allow thin (1$\mu$m) optical sections of fluorescent
samples to be captured. By imaging at a series of focal planes, stacks of
optical sections can be captured to generate a volumetric image and this can be
repeated over time to generate 3D+t images. The axial (z) resolution with these
techniques is typically around 5-fold less than the planar (xy) resolution so
the volumetric images are anisotropic. Confocal imaging provides 2-fold higer
resolution than 2-photon imaging but has a depth penetration limit of  150 um.
2-photon imaging can image all the way through a zebrafish embryo and has
reduced phototoxicity to the embryo and photobleaching of the fluorescent
labels.\\

Both acquisition techniques have been assembled in the Megasaon Lab from a
Zeiss 710 NLO (figure~\ref{fig:MicMegason}).


\begin{figure}[htb]
\begin{center}
\leavevmode
 \includegraphics[width=0.95\textwidth]{pictures/ConfocalPrinciple}
\end{center}
\caption{Confocal microscope principle
(source:\href{http://en.wikipedia.org/wiki/File:Confocalprinciple.svg}{Wikipedia
})}
\label{fig:ConfocalPrinciple}
\end{figure}

\begin{figure}[htb]
\begin{center}
\leavevmode
 \includegraphics[width=0.95\textwidth]{pictures/ConfocalZeissPrinciple}
\end{center}
\caption{2-photon confocal microscope principle (Zeiss documentation)}
\label{fig:Confocal2photonsPrinciple}
\end{figure}

\begin{figure}[htb]
\begin{center}
\leavevmode
 \includegraphics[width=0.95\textwidth]{pictures/PICmicroscope}
\end{center}
\caption{2-photon confocal microscope Zeiss 710 NLO in service in the Megason
Lab}
\label{fig:MicMegason}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{Image Processing}

After the embryos have been labeled and imaged, the final challenge of in toto
imaging is image processing. The goal of image processing is to track all the
cell movements and divisions to generate cell lineage trees, to define the
boundaries of all cells and their compartments (nucleus, cytoplasm, membrane,
extracellular space), and to quantize the level of fluorescence within each
cell and subcellular compartment.\\

In order to process all the acquired data, an image processing team of
is working on the next platform of microscopy images analysis: {\gofigure}.
Their goal is to create a very accessible "cross-platform"\footnote{In
computing, cross-platform, or multi-platform, is
an attribute conferred to computer software or computing methods and concepts
that are implemented and inter-operate on multiple operating systems and
hardware.}, "open-source"\footnote{of or
relating to or being computer software for which
the source code is freely available, but which use may be restricted by a
license.}, freely distributed, application with a high quality code, for
biologists to process their data and for computer scientists and image
processing specialists the opportunity to promote their methods.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\subsection{Biological data analysis}

Biologists acquire tremendous amount of data (see table~\ref{tab:DataSizes})
corresponding in three dimensional videos of zebra fishes embryos
(see figure~\ref{fig:fig:embryoConfocalSlice}).

\begin{figure}[htb]
\begin{center}
\leavevmode
\includegraphics[width=0.95\textwidth]{pictures/ConfocalPrinciple}
\end{center}
\caption{Confocal microscope principle (source:\href{http://en.wikipedia.org/wiki/File:Confocalprinciple.svg}{Wikipedia})}
\label{fig:ConfocalPrinciple}
\end{figure}

\begin{figure}[htb]
\begin{center}
\leavevmode
 \includegraphics[width=0.95\textwidth]{pictures/ConfocalZeissPrinciple}
\end{center}
\caption{2-photon confocal microscope principle (Zeiss documentation)}
\label{fig:Confocal2photonsPrinciple}
\end{figure}

\begin{figure}[htb]
\begin{center}
\leavevmode
\includegraphics[width=0.95\textwidth]{pictures/PICmicroscope}
\end{center}
\caption{2-photon confocal microscope Zeiss 710 NLO in service in the Megason Lab}
\label{fig:MicMegason}
\end{figure}

\clearpage

\section{Challenges of data sets to be processed}

\subsection{Large amount of data}
\subsection{tout ce qui est complique}

\section{Problematic}

Here What's the problem you want to solve

% \input{recherche/Recherche}

\chapter{State of the art}
\chapter{My Proposals}
\chapter{Results}


\chapter{Conclusion}
%
%
%
%\chapter{Conclusion du PFE}
%
%
%\section{Conclusion de la partie ingénierie}
%
%Durant ce PFE, j'ai travaillé sur des projets très variés.
%Ces projets étaient motivés par des objectifs scientifiques et nécessitaient chacun un haut niveau de technicité.
%
%J'ai eu l'occasion d'apprendre énormément,
%notamment dans le domaine des sciences informatiques,
%et du traitement de l'image.
%
%J'ai aussi découvert le milieu de la recherche, dont l'organisation est
%totalement différente de celle du milieu industriel.
%La hiérarchie, et les intérêts de chaque scientifique sont moins apparents que dans une entreprise et ce PFE a été riche en enseignements sur ce plan.
%Je suis parvenu à très bien m'intégrer dans l'équipe du Megason Lab en général,
%et à établir de bonnes relations de travail avec les divers scientifiques
%impliqués dans les projets sur lesquels j'ai travaillé.
%
%Cela m'a permis d'acquérir une base solide pour mener des recherches dans ce domaine.
%Après une importante phase d'apprentissage, je maitrise maintenant des librairies de renom
%(ITK, VTK,Qt), les systèmes unix, et la gestion de versions.
%
%J'ai pu aussi établir des contacts dans la communauté
%des traiteurs d'image par le biais de collaborations
%(Matthiew MacCormick), ou de d'"ateliers" (workshops) internationaux
%(participation à la NAMIC
%\footnote{La National Alliance for Medical Image Computing (NAMIC), est un réseau de professionnels de traitement d'images médicales. Ce réseau organise des conférences et des ateliers de travail ("workshop"), durant lesquels les scientifiques peuvent collaborer.}
%summer project week, au Massachusetts Institute of Technologies).
%
%
%\section{Projet professionnel}
%
%Ce PFE s'inscrit dans un projet professionnel construit durant ma scolarité à l'INSA de Lyon.
%Mon inscription à l'INSA de Lyon a été grandement motivée par l'ouverture de l'école à l'international. J'ai ainsi effectue mon stage ouvrier en Afrique du Sud. Profitant d'une première expérience professionnelle à l'international.
%
%Lors de ma seconde année a l'INSA de Lyon, j'ai choisi l'option SCiences et ANglais (SCAN).
%Cette filière regroupe les élèves ayant un niveau suffisant pour pouvoir suivre la formation généraliste en anglais.
%Elle s'accompagne d'une bourse pour un bref séjour linguistique dans une université étrangère.
%J'ai profité de ce financement pour aller au Trinity College en Irlande où
%j'ai pu avoir ma première expérience académique à l'étranger.
%
%J'ai ensuite choisi le département
%Génie Électrique qui permet de bénéficier d'un grand panel de compétences,
%notamment dans des domaines rattachés à l'électronique et l'informatique.
%
%
%Continuant l'expérience internationale, j'ai effectué mon stage industriel au
%Fraunhofer Institute, Center for Manufacturing Innovation, aux États Unis.
%J'ai été impliqué dans un projet de grande ampleur, financé par le département de la défense américain.
%Ce stage m'a énormément apporté en plus de l'expérience technique : il s'agissait d'une immersion dans la culture américaine, et j'ai eu une première expérience managériale en dirigeant une équipe d'ouvriers.
%C'est après ce stage que j'ai passé le test TOEIC validant un très bon niveau d'expression et de compréhension en anglais,
%avec un score de neuf cent quatre-vingt sur neuf cent quatre-vingt dix.
%
%
%Le PFE au Megason Lab possède des attraits indéniables :
%il s'agit d'un stage de recherche perfectionnant mes connaissances en traitement de l'image.
%Il s'agit aussi d'une expérience internationale dans une université renommée : Harvard Medical School.
%Ainsi, en plus de prendre plaisir à travailler en recherche, j'ouvre la porte à de nombreuses opportunités professionnelles.
%
%Comme je suis intéressé par les domaines techniques,
%je compte profiter de l'opportunité de poursuivre mes études pour obtenir un doctorat en mathématiques appliquées,
%au cours d'une thèse élaborée en collaboration avec le laboratoire CREATIS (INSA Lyon)
%et le laboratoire Megason (Harvard Medical School).
%Je passerai ainsi la moitié de mon temps en France, et l'autre moitié aux États Unis.
%En plus de pouvoir travailler sur un domaine passionnant,
%je pourrai ainsi bénéficier d'une équivalence avec un PhD (diplôme très reconnu à l'international). 
%
%Je compte ensuite mettre en valeur ma capacité a travailler dans un contexte international,
%dans des domaine à haute technicité pour trouver un emploi  dans le secteur privé.
%Je pourrai mettre en avant mes capacités en traitement du signal pour travailler dans le médical ou l'aéronautique.
%
%Mon ambition à long terme est de migrer vers des responsabilités managériales après avoir une très bonne maitrise de la technique.

\chapter*{Thanks}
\phantomsection
\addcontentsline{toc}{chapter}{Thanks}

I would like to thank, for their help and participation to my thesis, the following persons.

\begin{center}
Sean MEGASON\\
Kishore MOSALIGANTI\\
Arnaud GELAS\\
Olivier BERNARD\\
Rémy PROST\\
Matthiew MACCORMICK\\
Isabelle BLOCH\\
Nicolas RANNOU\\
Lydie SOUHAIT\\
Amelia GREEN\\
Nikolaus OBHOLZER\\
Ramil NOCHE\\
Raghav K. PADMANABHAN\\
Luis IBANEZ\\
\end{center}

And also the teams in Creatis LRMN, and in the Megason lab, for the great working ambiance I could find in these laboratories.

\appendix

%\section{The team}
%
%The Megason Lab's employee are post-doc researchers. Two domains of expertise are present : Biology and informatics.
%The biology team leads researches on the development of zebra fishes,
%and the computer team works on developing a program for visualization, segmentation, and tracking of cells adapted to microscopy data.
%
%{\small \begin{tabular*}{1.0\textwidth}{@{\extracolsep{\fill}} |  p{2.5cm} |  p{3.cm} | p{3.5cm} | c | }
%\hline Name & Status & Interest & Nationality \\ 
%\hline Sean Megason & Professor & Financing and managing the laboratory. & USA \\ 
%\hline Ramil Noche & Post-doctoral fellow & dynamics and function of gene regulatory networks. & Philipines \\ 
%\hline Fengzhu Xiong & Graduate student & cell differentiation mechanism to create neurons and skin cell. & Chine \\ 
%\hline Nikolaus Obholzer & Post-doctoral fellow  & Ear development and regeneration. &  Allemange \\ 
%\hline Paul Cowgill & Graduate student & Synthetic biology, noise and development. & USA \\ 
%\hline David Tulga & Graduate student & Cellular development. &  USA \\ 
%\hline Ian Swiburne & Post-doctoral fellow & Ear development. & USA \\ 
%\hline Andrea Tentner & Post-doctoral fellow  & Cell differentiation in spinal cord. & USA \\ 
%\hline Amelia Green & Post-doctoral fellow  & Variation and regulation in organ development. & Angleterre \\ 
%\hline Evan Schwab & Associate & Locations of genes responsible for mutations in the zebra fish ear using computational algorithm. & USA \\ 
%\hline Dante D'India & Technicien & Animal care. & USA \\ 
%\hline Arnaud Gelas & Research Engineer (senior) &  Developing {\gofigure}: Project management. &  France \\
%\hline Kishore Mosaliganti & Post-doctoral fellow  & Image processing algorithms, registration and segmentation. & India \\ 
%\hline Lydie Souhait & Research Engineer & Developing {\gofigure}: Responsible of the database (design, connexion, requests), and GUI & France \\
%\hline Nicolas Rannou & Research Engineer &  Developing {\gofigure}: Responsible of the visualization in {\gofigure} & France \\
%\hline Antonin Perrrot-Audet & Intern& Image processing : cells detection; developing {\gofigure} & France \\
%\hline 
%\end{tabular*} }

%\input{ingenierie/AnnexesIngenierie}

\bibliographystyle{plain}
\bibliography{AntoBib}

\end{document}
